### PR TITLE
[14.0] Take into account `github.ref` when doing upgrade-downgrade tests

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This is a backport of #10504 on `release-14.0`.

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: .github/workflows/static_checks_etc.yml [hunk 1]
```diff
@@ -59,85 +59,51 @@ jobs:
               - 'tools/**'
               - 'bootstrap.sh'
               - 'misc/git/hooks/asthelpers'
-            end_to_end:
+            docker:
```
Comment (deepthi) [inline .github/workflows/static_checks_etc.yml:62]: Were changes to this file intentional?

**Review by deepthi — APPROVED** [review]
Approving provisionally, see my comment

File: .github/workflows/static_checks_etc.yml (hunk 1 above)
Comment (frouioui) [reply .github/workflows/static_checks_etc.yml:62]: Yes, it was, I thought it made more sense as we are not using any end-to-end tests in this workflow. This filtering block is used for the docker checks.

**Comment by deepthi** [comment]
Let's create an issue to track the upgrade/downgrade test failures. Most of the failures are hangs during vttablet startup / vtgate shutdown, so I'm suspecting a problem with the test framework.

**Pull request merged**